app/routes.php

```diff
@@ -380,15 +380,17 @@ function( Application $app, Request $request ) use ( $ffFactory ) {
 );
 
 // Show a donation form with pre-filled payment values, e.g. when coming from a banner
-$app->get( 'donation/new', function ( Request $request ) use ( $ffFactory ) {
-	$amount = Euro::newFromFloat( ( new AmountParser( 'de_DE' ) )->parseAsFloat( $request->get( 'betrag', '' ) ) );
+$app->match( 'donation/new', function ( Application $app, Request $request ) use ( $ffFactory ) {
+	$amount = $request->get( 'betrag', $request->get( 'betrag_auswahl', '' ) );
+	$validationResult = ( $ffFactory->newPaymentDataValidator() )->validate( $amount, (string) $request->get( 'zahlweise', '' ) );
 
 	return new Response( $ffFactory->newDonationFormPresenter()->present(
-		$amount,
+		Euro::newFromFloat( ( new AmountParser( 'de_DE' ) )->parseAsFloat( $amount ) ),
```

Comment:
I was going to say "but what about the exception!", though then I realized you just moved this part. We still have some TODOs around the presentation specific parsing and formatting of values scattered around...